### PR TITLE
Indexer metrics ep2 for splunkenterprisereceiver

### PR DESCRIPTION
**Pull request by greatestusername**
**Description:** 
Adds additional indexer metrics to Splunk Enterprise receiver obtained from ad-hoc searches against Indexer or Cluster Manager instances.

Disables by default API endpoint searches that only return results for the specific instance which is having its API called. These can be re-enabled in config if those metrics are specifically required.

Generated tests and docs for these additional metrics

**Link to tracking Issue:** [30704](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/30704)

**Testing:** Tests generated with mdatagen

**Documentation:** Documentation generated with mdatagen

### CODE REVIEW

File: receiver/splunkenterprisereceiver/metadata.yaml [hunk 1]
```diff
@@ -30,81 +36,206 @@ metrics:
...
+      value_type: double 
+    attributes: [splunk.host]
+  splunk.scheduler.average.run.time:
+    enabled: true
+    description: Gauge tracking the average runtime of scheduled searches
+    unit: '{ms}'
+    gauge:
+      value_type: double 
+    attributes: [splunk.host]
+  splunk.indexer.raw.write.seconds:
```
Comment (atoulme) [inline receiver/splunkenterprisereceiver/metadata.yaml:102]: I think we don't put seconds in metric names usually

File: receiver/splunkenterprisereceiver/metadata.yaml [hunk 2]
```diff
@@ -30,81 +36,206 @@ metrics:
...
+    attributes: [splunk.index.name]
+  splunk.indexes.avg.size:
+    enabled: true
+    description: Gauge tracking the indexes and their average size (gb). *Note:** Search is best run against a Cluster Manager.
+    unit: Gb
+    gauge:
+      value_type: double 
+    attributes: [splunk.index.name]
+  splunk.indexes.avg.usage:
+    enabled: true
```
Comment (atoulme) [inline receiver/splunkenterprisereceiver/metadata.yaml:152]: you're adding those metrics as enabled by default, so please make sure to change your changelog to "breaking".

**Comment by atoulme** [comment]
Good stuff! Just some nits on metrics. I didn't check the scraper changes yet.

File: receiver/splunkenterprisereceiver/metadata.yaml (hunk 2 above)
Comment (greatestusername) [reply receiver/splunkenterprisereceiver/metadata.yaml:152]: Will do on these! Thanks for pointing this out!

Receiver isn't "wired in" yet so wasn't sure if `breaking` was appropriate.

File: receiver/splunkenterprisereceiver/metadata.yaml (hunk 1 above)
Comment (greatestusername) [reply receiver/splunkenterprisereceiver/metadata.yaml:102]: Thanks for the heads up! Didn't know this doc existed but probably should have :D 
https://github.com/open-telemetry/semantic-conventions/blob/main/docs/general/metrics.md#units

**Review by shalper2 — APPROVED** [review]

**Comment by greatestusername** [comment]
@MovieStoreGuy this should cover most all of the indexer metrics for this phase. Let me know if you notice anything required after you've given the PR a look? Thanks!

**Review by MovieStoreGuy — APPROVED** [review]

**Review by shalper2 — APPROVED** [review]

**Comment by greatestusername** [comment]
2 failures outside of my control in current tests: 
```
--- FAIL: TestGetNewToken (0.01s)
    --- FAIL: TestGetNewToken/Bad_payload_returned (0.01s)
        testing.go:1465: race detected during execution of test
    testing.go:1465: race detected during execution of test
FAIL
FAIL	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/bigipreceiver	0.303s
?   	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/bigipreceiver/internal/mocks	[no test files]
?   	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/bigipreceiver/internal/models	[no test files]
ok  	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/bigipreceiver/internal/metadata	1.098s
FAIL
```

Wil re-run on next base mergeupdate

**Comment by greatestusername** [comment]
le sigh :D 
```
2024/02/08 15:20:07 http: TLS handshake error from 127.0.0.1:60516: remote error: tls: bad certificate
2024/02/08 15:20:07 http: TLS handshake error from 127.0.0.1:60518: remote error: tls: bad certificate
2024/02/08 15:20:07 http: TLS handshake error from 127.0.0.1:60534: remote error: tls: bad certificate
2024/02/08 15:20:07 http: TLS handshake error from 127.0.0.1:60544: remote error: tls: bad certificate
2024/02/08 15:20:07 http: TLS handshake error from 127.0.0.1:60554: write tcp 127.0.0.1:43767->127.0.0.1:60554: use of closed network connection
FAIL
	github.com/open-telemetry/opentelemetry-collector-contrib/exporter/signalfxexporter	coverage: 88.4% of statements
FAIL	github.com/open-telemetry/opentelemetry-collector-contrib/exporter/signalfxexporter	8.832s
```

**Comment by greatestusername** [comment]
one more time!
```
[2024-02-08T16:19:29.200Z] ['error'] There was an error running the uploader: Error uploading to [https://codecov.io:](https://codecov.io/) Error: There was an error fetching the storage URL during POST: 503 - upstream connect error or disconnect/reset before headers. reset reason: connection failure
[2024-02-08T16:19:29.200Z] ['verbose'] The error stack is: Error: Error uploading to [https://codecov.io:](https://codecov.io/) Error: There was an error fetching the storage URL during POST: 503 - upstream connect error or disconnect/reset before headers. reset reason: connection failure
    at main (/snapshot/repo/dist/src/index.js)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
```

**Comment by greatestusername** [comment]
@jpkrohling or @TylerHelmuth anything else I need to do on this PR? Let me know! Thanks!

**Pull request merged**